provider/spark_test.go

```diff
@@ -525,8 +529,12 @@ func sparkTestInvalidTrainingSetDefs(t *testing.T, store OfflineStore) {
 		},
 	}
 	for name, def := range invalidDefs {
-		t.Run(name, func(t *testing.T) {
-			if err := store.CreateTrainingSet(def); err == nil {
+		nameConst := name
+		defConst := def
+		t.Run(nameConst, func(t *testing.T) {
+			t.Parallel()
+			time.Sleep(time.Second * 15)
```

Comment:
Whats this sleep doing?

Reply:
solution to avoid the throttling exception. Better solution would be to implement an exponential backoff retry function.

Reply:
Where does the parallelization start? Would this start them all, then they all wait 15 seconds, then start at the same time?